src/main/scala/com/softwaremill/realworld/articles/comments/CommentsService.scala

```diff
@@ -33,11 +33,21 @@ class CommentsService(
     _ <- commentsRepository.deleteComment(commentId)
   } yield ()
 
-  def getCommentsFromArticle(slug: String): Task[List[Comment]] =
-    for {
-      articleId <- articleIdBySlug(slug)
-      commentList <- commentsRepository.findComments(articleId)
-    } yield commentList
+  def getCommentsFromArticle(slug: String, emailOpt: Option[String]): Task[List[Comment]] = {
+    emailOpt match
+      case Some(email) =>
+        for {
+          userId <- userIdByEmail(email)
```

Comment:
**suggestion:** Just leaving this as a reminder of what we've discussed outside of GH: Please load userId into the session on login, to avoid querying for it every time you need it in the logic.

Reply:
I made the issue for this one